### ISSUE TITLE
Apply a Linter

### ISSUE THREAD

**Issue by brauliobo** [brigade/scss-lint#533]
I've just found your gem and it almost fits what I need!

I want to merge or append one selector into another. For example:
```scss
#chat {
  background: yellow;
}

#chat #buddy-list {
  border: 1px solid black;
}
```
Should be rewriten to:
```scss
#chat {
  background: yellow;
  #buddy-list {
    border: 1px solid black;
  }
}
```

But looking at the code and the command-line, I can't see that `scss-lint` is capable of applying linters' suggestions or generating a new SCSS file with the corrections all or some of the linters suggested.
Is that correct?

I would like to implement that, maybe a with method `apply!` on `Linter`.

**Comment by trotzig** [comment]
Hi @brauliobo! This was previously discussed in https://github.com/brigade/scss-lint/issues/343.

**Comment by brauliobo** [comment]
Could not resist and made a working version at #534. Please take a look at the code.
